<?xml version="1.0" encoding="UTF-8"?>

<launch>

  <!-- Arguments -->
  
  <!-- Replay Options -->
  <arg name="use_sim_time" default="true"/>
  <arg name="bag_post_fix" default="box_combined_tf"/>
  <arg name="output_folder" default=""/>
  <arg name="tf_bag_paths" default="[]"/>

  <!-- Parameters are here -->
  <arg name="config_path" default="" />

  <!-- Parameters are here -->
  <!-- ROS Param -->
  <rosparam file="$(arg config_path)" command="load"/>

  <rosparam param="tf_containing_bag_paths" subst_value="True">$(arg tf_bag_paths) </rosparam >

  <param name="/use_sim_time" value="$(arg use_sim_time)"/>

  <!-- Main Node -->
  <node name="box_tf_processor" pkg="box_auto" type="tf_processor_node" output="screen" required="True">
    <param name="output_folder" value="$(arg output_folder)"/>
    <param name="bag_post_fix" value="$(arg bag_post_fix)"/>
  </node>

</launch>